src/omnicore/omnicore.cpp

```diff
@@ -3607,309 +3624,106 @@ const CBitcoinAddress ExodusAddress()
-        newSP.fixed = true;
-        newSP.creation_block = newSP.update_block = chainActive[block]->GetBlockHash();
-
-        const unsigned int id = _my_sps->putSP(ecosystem, newSP);
-        update_tally_map(sender, id, nValue, BALANCE);
-      }
-      rc = 0;
-      break;
+    if (!interpret_Transaction()) {
+        return -98765;
     }
 
-    case MSC_TYPE_CREATE_PROPERTY_VARIABLE:
-    {
-      const char *p = step2_SmartProperty(step_rc);
-      if (0>step_rc) return step_rc;
-      if (!p) return (PKT_ERROR_SP -12);
-
-      step_rc = step3_sp_variable(p);
-      if (0>step_rc) return step_rc;
-
-      // check if one exists for this address already !
-      if (NULL != getCrowd(sender)) return (PKT_ERROR_SP -20);
-
-      // must check that the desired property exists in our universe
-      if (false == _my_sps->hasSP(property)) return (PKT_ERROR_SP -30);
-
-      if (0 == step_rc)
-      {
-        CMPSPInfo::Entry newSP;
-        newSP.issuer = sender;
-        newSP.txid = txid;
-        newSP.prop_type = prop_type;
-        newSP.num_tokens = nValue;
-        newSP.category.assign(category);
-        newSP.subcategory.assign(subcategory);
-        newSP.name.assign(name);
-        newSP.url.assign(url);
-        newSP.data.assign(data);
-        newSP.fixed = false;
-        newSP.property_desired = property;
-        newSP.deadline = deadline;
-        newSP.early_bird = early_bird;
-        newSP.percentage = percentage;
-        newSP.creation_block = newSP.update_block = chainActive[block]->GetBlockHash();
-
-        const unsigned int id = _my_sps->putSP(ecosystem, newSP);
-        my_crowds.insert(std::make_pair(sender, CMPCrowd(id, nValue, property, deadline, early_bird, percentage, 0, 0)));
-        PrintToLog("CREATED CROWDSALE id: %u value: %lu property: %u\n", id, nValue, property);  
-      }
-      rc = 0;
-      break;
+    if (obj_o && MSC_TYPE_TRADE_OFFER != type) {
+        return -777; // can't fill in the Offer object !
     }
 
-    case MSC_TYPE_CLOSE_CROWDSALE:
-    {
-    CrowdMap::iterator it = my_crowds.find(sender);
-
-      if (it != my_crowds.end())
-      {
-        // retrieve the property id from the incoming packet
-        memcpy(&property, &pkt[4], 4);
-        swapByteOrder32(property);
+    if (mdex_o && MSC_TYPE_METADEX != type) {
+        return -778; // can't fill in the MetaDEx object !
+    }
 
-        if (msc_debug_sp) PrintToLog("%s() trying to ERASE CROWDSALE for propid= %u=%X\n", __FUNCTION__, property, property);
+    switch (type) {
+        case MSC_TYPE_SIMPLE_SEND:
+            return logicMath_SimpleSend();
 
-        // ensure we are closing the crowdsale which we opened by checking the property
-        if ((it->second).getPropertyId() != property)
+        case MSC_TYPE_SEND_TO_OWNERS:
         {
-          rc = (PKT_ERROR_SP -606);
-          break;
+            // TODO: remove file stuff
+            boost::filesystem::path pathOwners = GetDataDir() / OWNERS_FILENAME;
```

Comment:
I can confirm that the information in "mastercore_owners.log" is definitely available via the ```getsto_MP``` call - suggest this can be removed safely.